primitives/arithmetic/src/traits.rs

```diff
@@ -65,8 +53,55 @@ impl<T:
 	CheckedShl + CheckedShr + CheckedAdd + CheckedSub + CheckedMul + CheckedDiv +
 	Saturating + PartialOrd<Self> + Ord + Bounded +
 	HasCompact + Sized
+> BaseArithmetic for T {}
+
+/// A meta trait for arithmetic.
+///
+/// Arithmetic types do all the usual stuff you'd expect numbers to do. They are guaranteed to
+/// be able to represent at least `u32` values without loss, hence the trait implies `From<u32>`
+/// and smaller ints. All other conversions are fallible.
+pub trait SimpleArithmetic:
```

Comment:
get out

Reply:
So, `SimpleArithmetic` is already somewhat not that expressive. But splitting this now into 3 traits who all are doing basically the same. 

We should up with something better.

1. Have one trait, probably just called `Arithmetic`.
2. `MiniArithmetic` is just a subset of `SimpleArithmetic` or? We probably can just merge `Arithmetic` and `MiniArithmetic`.
3. `SimpleArithmetic` should get some better name, IDK :D The name should give a clear indication what the difference of the trait is.

Reply:
1. Agree that the `Simple` prefix is not cool. 

2. Mini Arithmetic is _not_ a subset. it is actually a superset logically, since it has the same operations but can be any size, while Simple Arithmetic can be u32 or bigger. But this is not visible trait-wise. Neither can be composed of the other one, at least in the current format. 

Hence, I created the third one:

- `BasicArithmetic` is just operations with no assumption on size
- `Mini` and `Simple` can go on an _compose_ it + add limitation on size. 

Aside from the fact that `Simple` is not a nice prefix, I think this is quite nice. dk what you would say it is out of control.

Reply:
2. Yeah, I actually meant it the way you described it :D 

`Mini` isn't a nice prefix as well :P Maybe `U8Arithmetic` + `U32Arithmetic`? To make it clear on reading?

Reply:
ahh and the fact that the prefix `Mini` also doesn't make much sense anymore :D 

How about this 

`RuntimeArithmetic` for the one which is u32 and above, since we also use it for blocknumber etc.
`Arithmetic` for the one which is generic on size.